test/tests/functional/pbs_provisioning_enhancement.py

```diff
@@ -63,15 +63,32 @@ class TestProvisioningJob_Enh(TestFunctional):
 pbs.logmsg(pbs.LOG_DEBUG, "aoe=%s,vnode=%s" % (aoe,vnode))
 pbs.logmsg(pbs.LOG_DEBUG, "fake os provisioning script")
 e.accept(0)
+"""
+    reject_runjob_hook = """
+import pbs
+e = pbs.event()
+j = e.job
+pbs.logmsg(pbs.LOG_DEBUG, "job " + str(j) + " solution " + str(j.exec_vnode))
+e.reject()
 """
 
     def setUp(self):
+
+        if len(self.moms.values()) < 2:
+            self.skipTest("Provide at least 2 moms while invoking test")
+
         TestFunctional.setUp(self)
         self.momA = self.moms.values()[0]
         self.momB = self.moms.values()[1]
         self.hostA = self.momA.shortname
         self.hostB = self.momB.shortname
 
+        if self.hostA == self.server.shortname:
```

Comment:
I'm not sure i understand this, can you please add a comment to explain what's happening? Also, you could probably check for self.server.shortname before the first time you initialize these variables itself

Reply:
I'll add a comment

Reply:
I made this change because the test sets up aoe on momA that means it expects momA to be a remote mom. So I added this code to make sure that even if the first mom is local mom, the test swaps them and uses the correct mom for provisioning.